domain/schema/model/sql/0022-resource.sql

```diff
@@ -0,0 +1,109 @@
+CREATE TABLE resource_origin_kind (
+    id INT PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_resource_origin_name
+ON resource_origin_kind (name);
+
+INSERT INTO resource_origin_kind VALUES
+(0, 'uploaded'),
+(1, 'store');
+
+CREATE TABLE resource_state (
+    id INT PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_resource_state
+ON resource_state (name);
+
+INSERT INTO resource_state VALUES
+(0, 'available'),
+(1, 'potential');
+
+CREATE TABLE resource (
+    uuid TEXT NOT NULL PRIMARY KEY,
+    application_id TEXT NOT NULL,
+    name TEXT NOT NULL,
+    origin_kind_id INT NOT NULL,
+    state_id INT NOT NULL,
+    size INT,
+    fingerprint TEXT,
+    created_at TIMESTAMP NOT NULL,
+    CONSTRAINT fk_resource_application_id
+    FOREIGN KEY (application_id)
+    REFERENCES application (uuid),
+    CONSTRAINT fk_resource_name
+    FOREIGN KEY (name)
+    REFERENCES resource_meta (name),
+    CONSTRAINT fk_resource_origin_kind_id
+    FOREIGN KEY (origin_kind_id)
+    REFERENCES resource_origin_kind (id),
+    CONSTRAINT fk_resource_state_id
+    FOREIGN KEY (state_id)
+    REFERENCES resource_state (id)
+);
+
+CREATE UNIQUE INDEX idx_resource ON resource (application_id, name, state_id);
```

Comment:
We can have the same app/name with different states, but not more than one in the same state?

Reply:
Correct, there can be only one for application/unit and one for repository, however they will share the application id and resource name. State will differentiate.